examples/beacon-api-sidecar-fetcher/src/main.rs

```diff
@@ -0,0 +1,297 @@
+                    }
+                }
+            }
+            node_exit_future.await
+        })
+        .unwrap();
+}
+
+/// Our custom cli args extension that adds one flag to reth default CLI.
+#[derive(Debug, Clone, clap::Parser)]
+struct BeaconSidecarConfig {
+    /// Beacon Node http server address
+    #[arg(long = "cl.addr", default_value_t = IpAddr::V4(Ipv4Addr::LOCALHOST))]
+    pub cl_addr: IpAddr,
+    /// Beacon Node http server port to listen on
+    #[arg(long = "cl.port", default_value_t = 5052)]
+    pub cl_port: u16,
+}
+
+impl Default for BeaconSidecarConfig {
+    /// Default setup for lighthouse client
+    fn default() -> Self {
+        Self {
+            cl_addr: IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), // Equivalent to Ipv4Addr::LOCALHOST
+            cl_port: 5052,
+        }
+    }
+}
+
+impl BeaconSidecarConfig {
+    /// Returns the http url of the beacon node
+    pub fn http_base_url(&self) -> String {
+        format!("http://{}:{}", self.cl_addr, self.cl_port)
+    }
+
+    /// Returns the URL to the beacon sidecars endpoint (https://ethereum.github.io/beacon-APIs/#/Beacon/getBlobSidecars)
+    pub fn sidecar_url(&self, block_root: B256) -> String {
+        format!("{}/eth/v1/beacon/blob_sidecars/{}", self.http_base_url(), block_root)
+    }
+}
+
+/// SideCarError Handles Errors from both EL and CL
+#[derive(Debug, Error)]
+pub enum SideCarError {
+    #[error("Reqwest encountered an error: {0}")]
+    ReqwestError(Error),
+
+    #[error("There was an error grabbing the blob from the tx pool: {0}")]
+    TransactionPoolError(BlobStoreError),
+
+    #[error("400: {0}")]
+    InvalidBlockID(String),
+
+    #[error("404: {0}")]
+    BlockNotFound(String),
+
+    #[error("500: {0}")]
+    InternalError(String),
+
+    #[error("Network error: {0}")]
+    NetworkError(String),
+
+    #[error("Data parsing error: {0}")]
+    DeserializationError(String),
+
+    #[error("{0} Error: {1}")]
+    UnknownError(u16, String),
+}
+
+/// tracks the futures associated with retrieving blob data from the beacon client
+type BeaconFutures =
+    FuturesUnordered<Pin<Box<dyn Future<Output = Result<Vec<BlobTransaction>, SideCarError>>>>>;
+
+/// Wrapper struct for CanonStateNotifications
+pub struct MinedSidecarStream<St, P>
+where
+    St: Stream<Item = CanonStateNotification> + Send + Unpin + 'static,
+{
+    events: St,
+    pool: P,
+    beacon_config: BeaconSidecarConfig,
+    client: reqwest::Client,
+    pending_requests: BeaconFutures,
+    queued_actions: VecDeque<BlobTransaction>,
+}
+
+/// First checks if the Blobtranscation for a given EIP4844 is stored locally, if not attempts to
+/// retrieve it from the CL Layer
+impl<St, P> Stream for MinedSidecarStream<St, P>
+where
+    St: Stream<Item = CanonStateNotification> + Send + Unpin + 'static,
+    P: TransactionPoolExt + Unpin + 'static,
+{
+    type Item = Result<BlobTransaction, SideCarError>;
```

Comment:
I'd like to include some additional metadata about the block this was mined in, basically similar to the tx receipt,

Reply:
Created new struct BlockMetadata (we can add to this if needed)